packages/auto_mappr/lib/src/extensions/dart_type_extension.dart

```diff
@@ -11,16 +12,45 @@ extension DartTypeExtension on DartType {
       isDartCoreEnum ||
       isDartCoreSymbol;
 
-  bool isSameExceptNullability(DartType other) {
-    final thisName = element?.name;
-    final otherName = other.element?.name;
+  /// Checks name, generics, library
+  /// and nullability if [withNullability] is not set.
+  bool isSame(
+    DartType other, {
+    bool withNullability = false,
+  }) {
+    // Not the same type of type.
+    if ((this is InterfaceType) ^ (other is InterfaceType)) {
+      return false;
+    }
+
+    // TODO(generics): somehow compare if generics are the same?
```

Comment:
forgotten todo?

Reply:
Related to https://github.com/netglade/auto_mappr/issues/12.